doc/pmon/pmon-sensormon.md

```diff
@@ -0,0 +1,260 @@
+
+It is advised to monitor the sensor alarms and use that to debug and identify any issues in the system. In the event, a sensor crosses a high or low threshold, syslogs will be raised indicating the alarm.
+
+	FJul 27 08:26:32.561330 sonic WARNING pmon#sensormond: High voltage warning: VP0P75_CORE_NPU2 current voltage 880mV, high threshold 856mV
+
+The alarm condition will be visible in the CLI ouputs for sensor data and system health.
+
+	e.g 
+	root@sonic:/home/cisco# show platform voltage
+	          Sensor        Voltage    High TH    Low TH    Crit High TH    Crit Low TH    Warning          Timestamp
+	----------------  -------------  ---------  --------  --------------  -------------  ---------  -----------------
+	VP0P75_CORE_NPU2         880 mV        720       684             720            664       True  20230204 11:35:22
+	
+	root@sonic:/home/cisco# show system-health detail
+	System status summary
+  	...
+  	  Hardware:
+       Status: Not OK
+       Reasons: Voltage sensor VP0P75_CORE_NPU2 measurement 880 mV out of range (679,856)
+    	...
+    	VP0P75_CORE_NPU2            Not OK    voltage
+    	...
+
+
+##### Platform Sensors Configuration
+
+Sensormond will use the platform APIs for retrieving platform sensor information. However, for platforms with only file-system/sysfs based drivers, a simple implementation is provided wherein the platform can specify the sensor information for the board and any submodules (such as fabric cards) in a data file and Sensormond can use that for finding sensors and monitoring them. 
+
+The file system/Sysfs based platform sensor information can be provided using a yaml file. The yaml file shall have the following format.
+
+ 	
+	sensors.yaml
+      
+    voltage_sensors:
+      - name : <sensor name>
+          sensor: <sysfs path>
+          high_thresholds: [ <critical>, <major>, <minor> ]
+          low_thresholds: [ <critical>, <major>, <minor> ]
+        ...
+      
+    current_sensors:
+      - name : <sensor name>
+          sensor: <sysfs path>
+          high_thresholds: [ <critical>, <major>, <minor> ]
+          low_thresholds: [ <critical>, <major>, <minor> ]
+        ...
+      
+    <module_name>:
+      voltage_sensors:
+        - name: <sensor name>
+            sensor: <sysfs path>
+            high_thresholds: [ <critical>, <major>, <minor> ]
+            low_thresholds: [ <critical>, <major>, <minor> ]
+          ...
+ 
+      current_sensors:
+        - name: <sensor name>
+            sensor: <sysfs path>
+            high_thresholds: [ <critical>, <major>, <minor> ]
+            low_thresholds: [ <critical>, <major>, <minor> ]
+          ...
+
+
+
+
+##### PDDF Support
+
+SONiC PDDF provides a data driven framework to access platform HW devices. PDDF allows for sensor access information to be read from platform specific json files. PDDF support can be added for voltage and current sensors which can be retrieved by Sensormon.
```

Comment:
@bmridul in this HLD scope, are you including to parse the .json too if a platform choose PDDF approach?Why can't we use existing PDDF approach to read from .json?

Reply:
You can use  API OR Yaml file OR PDDF  approach. Sensormon will work with any of these approaches used for platform support.